src/FileIO.chpl

```diff
@@ -62,6 +62,78 @@ module FileIO {
         } 
     }
 
+    /*
+     * Ensure the file is closed, disregard errors
+     */
+    proc ensureClose(tmpf:file): bool {
+        var success = true;
+        try {
+            tmpf.close();
+        } catch {
+            success = false;
+        }
+        return success;
+    }
+
+    /*
+     * Indicates whether the filename represents a glob expression as opposed to
+     * an specific filename
+     */
+    proc isGlobPattern(filename: string): bool throws {
+        return filename.endsWith("*");
+    }
+
+    /*
+     * Generates a list of filenames to be written to based upon a file prefix,
+     * extension, and number of locales.
+     */
+    proc generateFilenames(prefix : string, extension : string, A) : [] string throws { 
+        // Generate the filenames based upon the number of targetLocales.
+        var filenames: [0..#A.targetLocales().size] string;
+        for i in 0..#A.targetLocales().size {
+            filenames[i] = generateFilename(prefix, extension, i);
+        }   
+        return filenames;
+    }
+
+    /*
+     * Generates a file name composed of a prefix, which is a filename provided by
+     * the user along with a file index and extension.
+     */
+    proc generateFilename(prefix : string, extension : string, idx : int) : string throws {
+        var suffix = '%04i'.format(idx);
+        return "%s_LOCALE%s%s".format(prefix, suffix, extension);
+    }
+
+    /*
+     * Generates an array of filenames to be matched in APPEND mode and to be
+     * checked in TRUNCATE mode that will warn the user that 1..n files are
+     * being overwritten.
+     */
+    proc getMatchingFilenames(prefix : string, extension : string) throws {
+        return glob("%s_LOCALE*%s".format(prefix, extension));    
+    }
+
+    /*
+     * Returns a tuple composed of a file prefix and extension to be used to
+     * generate locale-specific filenames to be written to.
+     */
+    proc getFileMetadata(filename : string) {
```

Comment:
It seems to me that "metadata" is an overloaded term, so maybe we could call this `getFileNameInfo` or something like that? (I don't particularly like that name, so maybe someone else has another idea?)

With Parquet files, metadata refers to file information such as number of elements in the file, row group size, etc., so I would be in favor of not mixing the term to mean both.

Reply:
I agree with you, but I didn't want to change the `proc` names while moving them.  Although if everyone is cool with it we could do renames at the same time.